assets/js/modules/analytics-4/components/dashboard/ActivationBanner/SetupBanner.stories.js

```diff
@@ -0,0 +1,142 @@
+	provideSiteInfo,
+} from '../../../../../../../tests/js/utils';
+import WithRegistrySetup from '../../../../../../../tests/js/WithRegistrySetup';
+import { MODULES_ANALYTICS_4 } from '../../../datastore/constants';
+import * as ga4Fixtures from '../../../../analytics-4/datastore/__fixtures__';
+
+const Template = () => <ActivationBanner />;
+
+export const NoPropertyNoTag = Template.bind( {} );
+NoPropertyNoTag.storyName = 'No GA4 Property - No Existing Tag';
+
+export const WithPropertyNoTag = Template.bind( {} );
+WithPropertyNoTag.storyName = 'Existing GA4 Property - No Existing Tag';
+WithPropertyNoTag.decorators = [
+	( Story ) => {
+		const setupRegistry = ( registry ) => {
+			registry.dispatch( MODULES_ANALYTICS_4 ).setMeasurementID(
+				// eslint-disable-next-line sitekit/acronym-case
+				ga4Fixtures.webDataStreams[ 0 ].webStreamData.measurementId
+			);
+		};
+
+		return (
+			<WithRegistrySetup func={ setupRegistry }>
+				<Story />
+			</WithRegistrySetup>
+		);
+	},
+];
+
+export const WithPropertyAndTag = Template.bind( {} );
+WithPropertyAndTag.storyName = 'Existing GA4 Property - Existing Tag';
+WithPropertyAndTag.decorators = [
+	( Story ) => {
+		const setupRegistry = ( registry ) => {
+			registry.dispatch( MODULES_ANALYTICS_4 ).setMeasurementID(
+				// eslint-disable-next-line sitekit/acronym-case
+				ga4Fixtures.webDataStreams[ 0 ].webStreamData.measurementId
+			);
+			registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetExistingTag(
+				// eslint-disable-next-line sitekit/acronym-case
+				ga4Fixtures.webDataStreams[ 0 ].webStreamData.measurementId
+			);
+		};
+
+		return (
+			<WithRegistrySetup func={ setupRegistry }>
+				<Story />
+			</WithRegistrySetup>
+		);
+	},
+];
+
+export const NoPropertyWithTag = Template.bind( {} );
+NoPropertyWithTag.storyName = 'No GA4 Property - Existing Tag';
+NoPropertyWithTag.decorators = [
+	( Story ) => {
+		const setupRegistry = ( registry ) => {
+			registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetExistingTag(
+				// eslint-disable-next-line sitekit/acronym-case
+				ga4Fixtures.webDataStreams[ 0 ].webStreamData.measurementId
+			);
+		};
+
+		return (
+			<WithRegistrySetup func={ setupRegistry }>
+				<Story />
+			</WithRegistrySetup>
+		);
+	},
+];
+
+export default {
+	title: 'Modules/Analytics4/ActivationBanner/SetupBanner',
+	decorators: [
+		( Story ) => {
+			const setupRegistry = ( registry ) => {
+				provideModules( registry, [
+					{
+						slug: 'analytics-4',
+						active: true,
+						connected: true,
+					},
+				] );
+
+				provideSiteInfo( registry );
+				provideModuleRegistrations( registry );
+
+				registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.receiveGetExistingTag( null );
+			};
+
+			setTimeout( async () => {
+				await waitForElement( () =>
+					document.querySelector(
+						'a.googlesitekit-notification__cta'
+					)
+				);
+				document
+					.querySelector( 'a.googlesitekit-notification__cta' )
+					.click();
+			}, 0 );
```

Comment:
Why do we need this? Will the banner work without this?

Reply:
@eugene-manuilov Since the `<SetupBanner />` is displayed based on a local component state inside `<ActivationBanner />` and that state is only updated upon clicking on the CTA button on the banners, we need to simulate a click on the CTA to update the state inside `<ActivationBanner />` and display `<SetupBanner />`.

Reply:
We don't need to use the `ActivationBanner` component here since it is intended for the setup banner only. Let's update the template to be as following and remove this timeout workaround:

```
import SetupBanner from './SetupBanner';

const Template = () => <SetupBanner ... />;
```